internal/handlers/middlewares.go

```diff
@@ -74,3 +80,124 @@ func GzipCompress(h http.HandlerFunc) http.HandlerFunc {
 		h.ServeHTTP(ow, r)
 	}
 }
+
+var (
+	// TODO вынести данные переменные в переменные окружения
+	hashKey  = []byte("very-secret-key")  // Симметричный ключ для подписи
+	blockKey = []byte("a-lot-secret-key") // Симметричный ключ для шифрования
+	sCookie  = securecookie.New(hashKey, blockKey)
+)
+
+// ValidateUserUID - декоратор валидации JWT токена.
+func ValidateUserUID(cookieValue string) (string, bool) {
+	var userUID string
+	err := sCookie.Decode("userUID", cookieValue, &userUID)
+	if err == nil {
+		// Кука существует и проходит проверку, продолжаем выполнение следующего обработчика
+		logrus.Printf("Existing valid user ID: %s", userUID)
+		return userUID, true
+	}
+	return "", false
+}
+
+// CheckSignedCookie для подписанной куки с идентификатором пользователя.
+func CheckSignedCookie(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Попытка получить куку
+		cookie, err := r.Cookie("userUID")
+
+		if err == nil {
+			// Проверка и декодирование куки
+			userUID, isValid := ValidateUserUID(cookie.Value)
+			if isValid {
+				// Кука существует и проходит проверку, продолжаем выполнение следующего обработчика
+				logrus.Printf("Existing valid user ID: %s", userUID)
+				h.ServeHTTP(w, r)
+				return
+			} else {
+				logrus.Printf("Wrong UserUID: %s", cookie.Value)
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+		}
+
+		h.ServeHTTP(w, r)
```

Comment:
Стоит сразу сделать проверку ошибки. Это более наглядно и позволяет сделать обработку ошибок более читаемой.
```suggestion
	if err != nil {
	    h.ServeHTTP(w, r)
	}
	
	// Проверка и декодирование куки
	userUID, isValid := ValidateUserUID(cookie.Value)
	if isValid {
		// Кука существует и проходит проверку, продолжаем выполнение следующего обработчика
		logrus.Printf("Existing valid user ID: %s", userUID)
		h.ServeHTTP(w, r)
		return
	} else {
		logrus.Printf("Wrong UserUID: %s", cookie.Value)
		w.WriteHeader(http.StatusUnauthorized)
		return
	}

	h.ServeHTTP(w, r)
```